api/src/client/ApiFormationHelpers.ts

```diff
@@ -212,15 +221,17 @@ export const makePostBody = (
           formationFormData.businessStartDate,
           defaultDateFormat
         ).format(formationApiDateFormat),
-        MainAddress: {
-          Address1: formationFormData.addressLine1,
-          Address2: formationFormData.addressLine2,
-          City: formationFormData.addressMunicipality?.name ?? formationFormData.addressCity ?? "",
-          State: formationFormData.addressState?.name,
-          Province: formationFormData.addressProvince,
-          Zipcode: formationFormData.addressZipCode,
-          Country: formationFormData.addressCountry,
-        },
+        MainAddress: isStartingBusinessAddressPartiallyEmpty()
+          ? undefined
+          : {
+              Address1: formationFormData.addressLine1,
+              Address2: formationFormData.addressLine2,
+              City: formationFormData.addressMunicipality?.name ?? formationFormData.addressCity ?? "",
+              State: formationFormData.addressState?.name,
+              Province: formationFormData.addressProvince,
+              Zipcode: formationFormData.addressZipCode,
+              Country: formationFormData.addressCountry,
+            },
```

Comment:
[dust] This works, and we do have other similar logic in this object executed inline, but I think I'd opt for moving this all into a single function.

```suggestion
        MainAddress: getBusinessAddress()
```

Reply:
updated